samples/apps/copilot-chat-app/webapi/Controllers/ChatHistoryController.cs

```diff
@@ -95,18 +106,28 @@ public async Task<IActionResult> GetChatSessionByIdAsync(Guid chatId)
     /// The pattern matches two GUIDs in canonical textual representation separated by a period.
     /// </summary>
     /// <param name="userId">The user id.</param>
+    /// <returns>A list of chat sessions. An empty list if the user is not in any chat session.</returns>
     [HttpGet]
     [Route("chatSession/getAllChats/{userId:regex(([[a-z0-9]]+-)+[[a-z0-9]]+\\.([[a-z0-9]]+-)+[[a-z0-9]]+)}")]
     [ProducesResponseType(StatusCodes.Status200OK)]
-    [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     public async Task<IActionResult> GetAllChatSessionsAsync(string userId)
     {
-        var chats = await this._chatSessionRepository.FindByUserIdAsync(userId);
-        if (chats == null)
+        var chatParticipants = await this._chatParticipantRepository.FindByUserIdAsync(userId);
```

Comment:
I might be biased as how I usually do RBAC with objects (e.g. filter chat sessions that aren't associated with the user). The logic here looks different (different doesn't mean anything bad), could you explain more with comments? Why are we getting a list of users (participants) from a given user id, and find chats sessions per users (participants)?

Reply:
This is a good point. 
I guess the name `ChatParticipant` causes confusion here. 

User != Participant.

`ChatParticipant` contains the information of a participant in a particular chat session. One user can be multiple `ChatParticipant`s in different chat sessions.

To get all the chat sessions a user has access to, we need to first find all the `ChatParticipant`s that belongs to the user.

Reply:
got it. recommend adding inline comments to explain might help others understand the thoughts.